### ISSUE TITLE
Upgrade to 2.0.2 causes amr claim to default to external

### ISSUE THREAD

**Issue by chrisdrobison** [IdentityServer/IdentityServer4#1732]
- [ X] I read and understood how to enable [logging](https://identityserver4.readthedocs.io/en/release/topics/logging.html)

### Issue / Steps to reproduce the problem

To highlight, we have a project that hosts the identity server as well as an identity server admin API so to speak that allows us to create clients, api and identity resources with an angular app. So of the important code config tidbits are as follows:

```
services.AddAuthorization(options =>
{
    options.AddPolicy(Constants.AdminApiPolicy, policyBuilder => policyBuilder
        .RequireClaim(JwtClaimTypes.Role, Constants.AdminRole)
        .RequireClaim(JwtClaimTypes.AuthenticationMethod, OidcConstants.AuthenticationMethods.Password)
        .RequireScope("xactware.identityserver.adminclient")
        .AddAuthenticationSchemes(OidcConstants.AuthenticationSchemes.AuthorizationHeaderBearer));
});
```

```
services.AddAuthentication()
    .AddIdentityServerAuthentication(options =>
    {
        options.Authority = Configuration["idsrv:admin:authority"];
        options.RequireHttpsMetadata = false;
        options.JwtBearerEvents = new JwtBearerEvents
        {
            OnTokenValidated = async context =>
            {
                var adminRepo = context.HttpContext.RequestServices.GetService<AdminUserRepository>();
                var user = await adminRepo.FindUserById(context.Principal
                    .FindFirst(JwtClaimTypes.Subject)
                    .Value);
                if (user != null)
                {
                    context.Principal.AddIdentity(
                        new ClaimsIdentity(new[] {new Claim(JwtClaimTypes.Role, Constants.AdminRole)}));
                }
            }
        };
    });
```
We inject the admin role claim if the user is in a list we maintain. Other settings include adding identity server. So this all used to work with 2.0.1. With 2.0.2 we are getting the following in the logs. Changing the version of the AccessTokenValidation library doesn't have any effect. So it appears to be isolated to whatever changed with IdentityServer4 2.0.2.

### Relevant parts of the log file

```
2017-11-09 22:10:52.914 -07:00 [Information] AuthenticationScheme: "BearerIdentityServerAuthenticationJwt" was successfully authenticated.
2017-11-09 22:10:52.915 -07:00 [Information] AuthenticationScheme: "Bearer" was successfully authenticated.
2017-11-09 22:10:52.922 -07:00 [Information] Authorization failed for user: "330".
2017-11-09 22:10:52.927 -07:00 [Information] Authorization failed for the request at filter '"Microsoft.AspNetCore.Mvc.Authorization.AuthorizeFilter"'.
2017-11-09 22:10:52.928 -07:00 [Information] Executing ForbidResult with authentication schemes (["Bearer"]).
2017-11-09 22:10:52.931 -07:00 [Information] AuthenticationScheme: "Bearer" was forbidden.
```

**Comment by leastprivilege** [comment]
Hard to tell - simplify your authZ policy - have a look at the claims of the user.

That should give you more info.

**Comment by chrisdrobison** [comment]
OK, that helped. It looks like in 2.0.1, the amr claim is coming back as `pwd`, where as in 2.0.2 it is coming back as `external`. I'll go look at the sign-in to see what is happening there.

**Comment by leastprivilege** [comment]
That would be a bug. I will also have a look

**Comment by leastprivilege** [comment]
fixed in 2.0.3
